spec/api/ldapsync_spec.rb

```diff
@@ -12,64 +12,172 @@
+      end
     end
-  end
 
-  describe "#ldap_sync_now" do
-    let(:dry_run) { false }
-    it "POSTs /sync with dry_run set to false" do
-      api.ldap_sync_now('default', 'application/json', false)
+    describe 'LdapSyncJob#delete' do
+      let(:job){ Conjur::LdapSyncJob.new(api, 'job-id', false, 'connect', 'running') }
+      let(:url){ "#{appliance_url}/ldap-sync/jobs/#{job.id}" }
+
+      it 'calls delete on the job resource' do
+        expect_request(
+            url: url,
+            method: :delete,
+            headers: credentials[:headers]
+        )
+        job.delete
+      end
+
     end
-  end
 
-  describe "#ldap_sync_now" do
-    let(:dry_run) { true }
-    it "POSTs /sync with truthy dry_run value" do
-      api.ldap_sync_now('default', 'application/json', 1)
+    describe 'LdapSyncJob#output' do
+
+      let(:response){ double('response', code: response_status.to_s) }
+
+      let(:chunks){[
+          'id:', " 0\n",
+          "event: foo\ndata: {\"message\": \"hello\"",  "}\n\n",
+          'id: 1', "\nevent: result\ndata: {\"ok\":false,\"error\":{\"message\":\"Cannot resolve host ldapserver\"}}\n\n"
+      ]}
+
+      let(:job){
+        Conjur::LdapSyncJob.new(api, 'job-id', false, 'connect', 'running')
+      }
+
+      let(:url){ "#{appliance_url}/ldap-sync/jobs/#{job.id}/output" }
+
+      before do
+        expect_request(
+            url: url,
+            method: :get,
+            headers: credentials[:headers].dup.merge(accept: 'text/event-stream'),
+            block_response: instance_of(Proc)
+        ) do |opts|
+          opts[:block_response].call response
+        end
+        expect(response).to receive(:read_body) do |&blk|
+          chunks.each{ |c| blk[c] }
+        end
+      end
+
+      context 'when the request succeeds' do
+
+        let(:response_status){ 200 }
+
+        context 'when given a block' do
+          it 'is called with each event' do
+            blocked = []
+            events = job.output do |event|
+              blocked << event
+            end
+            expect(blocked.length).to eq(2)
+            expect(blocked).to eq(events)
+          end
+        end
+
+        context 'when not given a block' do
+          it 'returns the event data' do
+            events = job.output
+            expect(events[0]['message']).to eq('hello')
+            expect(events[1]['ok']).to be_falsey
+            expect(events[1]['error']['message']).to eq('Cannot resolve host ldapserver')
+          end
+        end
+      end
+
+      context 'when the request fails' do
+        let(:response_status){ 422 }
+
+        it 'calls response.error!' do
+          expect(response).to receive(:error!)
+          job.output
+        end
+      end
     end
-  end
 
-  describe "#ldap_sync_now" do
-    let(:dry_run) { false }
-    it "POSTs /sync with falsey dry_run value" do
-      api.ldap_sync_now('default', 'application/json', nil)
+    describe '#ldap_sync_now' do
```

Comment:
No this tests synchronous sync.

On Wed, Aug 10, 2016, 10:27 AM Kurt Stutsman <notifications@github.com>
wrote:

> In spec/api/ldapsync_spec.rb
> <https://github.com/conjurinc/api-ruby/pull/83#discussion_r74267439>:
>
> >
> > -  describe "#ldap_sync_now" do
> > -    let(:dry_run) { false }
> > -    it "POSTs /sync with falsey dry_run value" do
> > -      api.ldap_sync_now('default', 'application/json', nil)
> > +    describe '#ldap_sync_now' do
>
> Don't you need to take out these tests until you add support for /sync?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/conjurinc/api-ruby/pull/83/files/0e7fd5800641eb2b4148cf4234f55b13a4bb6e30#r74267439>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABK82N2Ur_nzdXVMuwSkZYCoL-Fvf4bjks5qee3_gaJpZM4JgfLo>
> .
>